src/server/auth/server/util.go

```diff
@@ -32,34 +32,18 @@ func (a *apiServer) CheckRepoIsAuthorizedInTransaction(txnCtx *txncontext.Transa
 // CheckResourceIsAuthorizedInTransaction returns an error if the subject/user doesn't have permission in `p` on the `resource`
 func (a *apiServer) checkResourceIsAuthorizedInTransaction(txnCtx *txncontext.TransactionContext, resource *auth.Resource, p ...auth.Permission) error {
 	me, err := txnCtx.WhoAmI()
-	if errors.Is(err, auth.ErrNotActivated) {
-		return nil
-	}
-
-	req := &auth.AuthorizeRequest{Resource: resource, Permissions: p}
-	resp, err := a.AuthorizeInTransaction(txnCtx, req)
 	if err != nil {
+		// FIXME stop swallowing ErrNotActivated
+		// Why are we swallowing? So that code that calls into this can not worry about auth activeness.
+		// However, we should let the code calling this explicitly handle it.
```

Comment:
Are you sure about that? That would mean that each call site would end up looking like:

```go
if err := srv.checkResourceIsAuthorizedInTransaction(…); err != nil {
  if errors.Is(err, auth.ErrNotActivated) {
    return nil
  }
}
```

which seems a bit much. And error-prone too. Maybe `WhoAmI` should not return an error when auth is inactive? That sounds iffy too …